src/modules/flow/json/json.c

```diff
@@ -90,15 +120,72 @@ create_sub_json(struct sol_blob *parent, struct sol_json_scanner *scanner, struc
 }
 
 static bool
-json_object_search_for_token(struct sol_json_scanner *scanner, const char *key, struct sol_json_token *value)
+_get_next_slice(const struct sol_str_slice path, struct sol_str_slice *slice)
```

Comment:
This function is pretty tough to review. Maybe you could use the FBP lexer/parser as an inspiration?